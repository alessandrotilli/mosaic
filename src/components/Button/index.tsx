import React, { FC, ReactElement } from "react";
import MUIButton from "@material-ui/core/Button";
import Icon from "../Icon";
import { ButtonIconPosition, IButtonIcon, IButton, ButtonVariants } from "../../types/Button";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

interface IMUIButtonIcon {
  endIcon?: ReactElement;
  startIcon?: ReactElement;
}

const getIcons = (dataCy: string, iconConfig?: IButtonIcon): IMUIButtonIcon => {
  const muiIcon = {};
  if (!iconConfig) {
    return muiIcon;
  }

  const { name, position } = iconConfig;
  const icon = <Icon dataCy={`${dataCy}-icon`} name={name} />;
  return position === ButtonIconPosition.right
    ? {
        endIcon: icon,
      }
    : { startIcon: icon };
};

export const DATA_CY_DEFAULT = "button";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [{ name: "label", type: "string" }];

// TODO: handle color
const Button: FC<IButton> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  elevated = false,
  icon = undefined,
  label,
  onClick,
  variant = ButtonVariants.contained,
}) => {
  return (
    <MUIButton
      color="primary"
      data-cy={dataCy}
      disabled={disabled}
      disableElevation={!elevated}
      onClick={onClick}
      variant={variant}
      {...getIcons(dataCy, icon)}
    >
      {label}
    </MUIButton>
  );
};

export const ButtonWithProps = Button;

export default localized(Button, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
