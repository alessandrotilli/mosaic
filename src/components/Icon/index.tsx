import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
