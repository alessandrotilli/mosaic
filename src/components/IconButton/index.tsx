import React, { FC, useCallback, useMemo } from "react";
import MUIIconButton from "@mui/material/IconButton";

import { IIconButton, IIconButtonSubpart } from "../../types/IconButton";
import { getComposedDataCy, ISubpartMap, suppressEvent } from "../../utils";
import Adornment, { ADORNMENT_SUBPARTS } from "../Adornment";
import IconWrapper from "../IconWrapper";

export const DATA_CY_DEFAULT = "icon-button";

export const SUBPARTS_MAP: ISubpartMap<IIconButtonSubpart> = {
  ...ADORNMENT_SUBPARTS,
  icon: {
    label: "Icon",
  },
};

const IconButton: FC<IIconButton> = ({
  badge,
  dataCy = DATA_CY_DEFAULT,
  icon,
  onClick,
  disabled = false,
  rotate = false,
  size = "medium",
  style,
  tooltip,
}) => {
  const adornmentDataCy = useMemo(() => getComposedDataCy(dataCy, SUBPARTS_MAP.adornment), [dataCy]);

  const iconDataCy = useMemo(() => getComposedDataCy(dataCy, SUBPARTS_MAP.icon), [dataCy]);

  const onClickHandler = useCallback(
    (event: any) => {
      suppressEvent(event);
      onClick();
    },
    [onClick]
  );

  return (
    <Adornment dataCy={adornmentDataCy} badge={badge} tooltip={tooltip}>
      <MUIIconButton color="inherit" data-cy={dataCy} disabled={disabled} onClick={onClickHandler} style={style}>
        <IconWrapper dataCy={iconDataCy} icon={icon} rotate={rotate} size={size} />
      </MUIIconButton>
    </Adornment>
  );
};

export default IconButton;
