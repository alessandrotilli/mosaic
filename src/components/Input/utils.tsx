import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
