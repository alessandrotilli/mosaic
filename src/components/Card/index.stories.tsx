import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, text } from "@storybook/addon-knobs";

import { Icons } from "../../types/Icon";
import { getAllComposedDataCy } from "../../utils";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";
import Button from "../Button";
import IconButton from "../IconButton";
import Typography from "../Typography";

import Card, { CardWithProps, DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, SUBPARTS_MAP } from ".";

export default {
  title: "Card",
  component: CardWithProps,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/Card",
      component: "Card",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
        dataCyShortcut: DATA_CY_SHORTCUT,
        subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
      },
      localizableProps: LOCALIZABLE_PROPS,
    }),
  },
};

export const Canvas = () => (
  <Card
    actions={[
      <Button label="Share" onClick={action("Share Button")} />,
      <IconButton icon={Icons.add} onClick={action("Add action")} />,
    ]}
    collapsible={
      <Typography>
        Traditional mosaics are made of small, flat, roughly square pieces of stone or glass of different colors, known
        as tesserae. Some floor mosaics are made of small rounded pieces of stone and called pebble mosaics. Mosaic
        skinning (covering objects with mosaic glass) is done with thin enameled glass and opaque stained glass. Modern
        mosaic art is made from any material in any size ranging from carved stone, bottle caps, and found objects.
      </Typography>
    }
    dataCy={text("dataCy", DATA_CY_DEFAULT)}
    icon={select("icon", Icons, Icons.add)}
    loading={boolean("loading", false)}
    localized={boolean("localized", false)}
    subtitle={text("subTitle", "Mosaic Card Subtitle")}
    title={text("title", "Mosaic Card")}
    unmountCollapsible={boolean("unmountCollapsible", false)}
  >
    <Typography>
      A mosaic is an artistic picture or design made out of any materials assembled together. Mosaic are used as
      decoration. Architects use mosaic murals for kitchen backsplash, shower wall and entry floor art. Mosaic Craft
      items are used as home decor. Cities often decorate public places such as parks with mosaic murals and sculptures.
    </Typography>
  </Card>
);

export const Basic = () => (
  <StoriesWrapper>
    <Card title="Mosaic">
      <Typography>A mosaic is an artistic picture or design made out of any materials assembled together.</Typography>
    </Card>
  </StoriesWrapper>
);

export const Collapsible = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      collapsible={<Typography children="Discover our catalog!" />}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    >
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);

export const Loading = () => (
  <StoriesWrapper>
    <Card icon={Icons.add} loading title="Mosaic">
      <Typography>A mosaic is an artistic picture or design made out of any materials assembled together.</Typography>
    </Card>
  </StoriesWrapper>
);

export const Localized = () => (
  // IntlProviderMock simulates external IntlProvider context
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <Card icon={Icons.business} localized title={MessageMock.title} subtitle={MessageMock.subtitle} />
    </IntlProviderMock>
  </StoriesWrapper>
);

export const WithAvatar = () => (
  <StoriesWrapper>
    <Card icon={Icons.business} subtitle="The best in town for components" title="Mosaic Shop">
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);

export const WithActions = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    >
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);
