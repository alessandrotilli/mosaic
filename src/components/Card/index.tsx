import React, { cloneElement, FC, Fragment, useMemo, useState } from "react";
import {
  Card as MUICard,
  CardActions as MUICardActions,
  CardContent as MUICardContent,
  CardHeader as MUICardHeader,
  Collapse as MUICollapse,
  useTheme,
} from "@material-ui/core";
import { Skeleton as MUISkeleton } from "@material-ui/lab";

import { ICard } from "../../types/Card";
import { Icons } from "../../types/Icon";
import { getComposedDataCy } from "../../utils";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";
import Avatar from "../Avatar";
import IconButton from "../IconButton";
import Typography from "../Typography";

export const DATA_CY_DEFAULT = "card";
export const DATA_CY_SHORTCUT = "title";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "title", type: "string" },
  { name: "subtitle", type: "string" },
];

export const SUBPARTS_MAP = {
  avatar: {
    label: "Avatar",
  },
  title: {
    label: "Title",
  },
  subtitle: {
    label: "Subtitle",
  },
  content: {
    label: "Content",
  },
  collapse: {
    label: "Collapse",
  },
  collapsibleContent: {
    label: "Collapsible Content",
  },
};

const Card: FC<ICard> = ({
  actions = [],
  children,
  collapsible,
  dataCy = DATA_CY_DEFAULT,
  icon,
  loading = false,
  style,
  subtitle,
  title,
  unmountCollapsible = false,
}) => {
  const [expanded, setExpanded] = useState(false);
  const theme = useTheme();

  const cardHeader = useMemo(
    () => (
      <MUICardHeader
        avatar={
          icon && <Avatar dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.avatar)} icon={icon} loading={loading} />
        }
        disableTypography
        title={
          <Typography
            bottomSpacing={false}
            dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.title)}
            loading={loading}
            truncated
            variant="title"
          >
            {title}
          </Typography>
        }
        subheader={
          <Typography
            bottomSpacing={false}
            dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.subtitle)}
            loading={loading}
            truncated
            variant="caption"
          >
            {subtitle}
          </Typography>
        }
      />
    ),
    [dataCy, icon, loading, subtitle, title]
  );

  return (
    <MUICard
      data-cy={dataCy}
      style={{ margin: `${theme.spacing(2)}px`, width: `calc(100% - ${theme.spacing(4)}px)`, ...style }}
    >
      {cardHeader}
      <MUICardContent
        data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.content)}
        style={{ padding: `${theme.spacing(1)}px ${theme.spacing(2)}px` }}
      >
        {loading ? <MUISkeleton height={`${theme.spacing(16)}px`} /> : children}
      </MUICardContent>
      {!loading && (
        <Fragment>
          <MUICardActions
            disableSpacing
            style={{
              alignItems: "center",
              display: "flex",
              justifyContent: "space-between",
              padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
            }}
          >
            {actions.length > 0 && (
              <div style={{ alignItems: "center", display: "flex" }}>
                {actions.map((action, index) => (
                  <div key={`card-action-${index}`} style={{ marginRight: `${theme.spacing(2)}px` }}>
                    {cloneElement(action)}{" "}
                  </div>
                ))}
              </div>
            )}
            {collapsible && (
              <IconButton
                dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapse)}
                icon={expanded ? Icons.up : Icons.down}
                onClick={() => setExpanded(!expanded)}
              />
            )}
          </MUICardActions>
          {collapsible && (
            <MUICollapse in={expanded} timeout="auto" unmountOnExit={unmountCollapsible}>
              <MUICardContent
                data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapsibleContent)}
                style={{ padding: `${theme.spacing(1)}px ${theme.spacing(2)}px` }}
              >
                {collapsible}
              </MUICardContent>
            </MUICollapse>
          )}
        </Fragment>
      )}
    </MUICard>
  );
};

export const CardWithProps = Card;

export default localized(Card, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
