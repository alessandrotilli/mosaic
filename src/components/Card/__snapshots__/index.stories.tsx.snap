// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surfaces/Card Actions smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <div style="align-items: center; display: flex;">
      <div style="margin-right: 16px;">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
                data-cy="button"
        >
          <span class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon">
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="OpenInNewIcon"
                 data-cy="button-icon"
            >
              <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z">
              </path>
            </svg>
          </span>
          Gallery
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card Collapsible play-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <button class="MuiButtonBase-root Mui-focusVisible MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
            data-cy="card-collapse"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="KeyboardArrowUpIcon"
           data-cy="card-collapse-icon"
      >
        <path d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z">
        </path>
      </svg>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
        <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 41px; height: 41px; top: -0.5px; left: -0.5px;"
        >
          <span class="MuiTouchRipple-child MuiTouchRipple-childPulsate">
          </span>
        </span>
        <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 41px; height: 41px; top: -0.5px; left: -0.5px;"
        >
          <span class="MuiTouchRipple-child MuiTouchRipple-childLeaving">
          </span>
        </span>
      </span>
    </button>
  </div>
  <div class="MuiCollapse-root MuiCollapse-vertical css-1fkrdiv-MuiCollapse-root"
       style="min-height: 0px; height: 35px; transition-duration: 191ms;"
  >
    <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
      <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
        <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
             data-cy="card-collapsible-content"
             style="padding: 8px 16px;"
        >
          Discover mosaics from all around the globe!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card CollapsibleActions smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <div style="align-items: center; display: flex;">
      <div style="margin-right: 16px;">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-zddlty-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
                data-cy="button"
        >
          <span class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon">
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="OpenInNewIcon"
                 data-cy="button-icon"
            >
              <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z">
              </path>
            </svg>
          </span>
          Gallery
          <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
          </span>
        </button>
      </div>
    </div>
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
            data-cy="card-collapse"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="KeyboardArrowDownIcon"
           data-cy="card-collapse-icon"
      >
        <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z">
        </path>
      </svg>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </button>
  </div>
  <div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
       style="min-height: 0px;"
  >
    <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
      <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
        <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
             data-cy="card-collapsible-content"
             style="padding: 8px 16px;"
        >
          Discover mosaics from all around the globe!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card Icon smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar">
      <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
           data-cy="card-avatar"
           style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="AppsIcon"
             data-cy="card-avatar-icon"
        >
          <path d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z">
          </path>
        </svg>
      </div>
    </div>
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;

exports[`Surfaces/Card Loading smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar">
      <span class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto css-1gtw0ls-MuiSkeleton-root">
        <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
             data-cy="card-avatar-loading"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="PersonIcon"
          >
            <path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
            </path>
          </svg>
        </div>
      </span>
    </div>
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title-loading"
      >
        <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root">
        </span>
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle-loading"
      >
        <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root">
        </span>
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
          style="height: 128px;"
    >
    </span>
  </div>
</div>
`;

exports[`Surfaces/Card Localized smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="locale.title"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="locale.title-title"
      >
        Titolo
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="locale.title-subtitle"
      >
        Sottotitolo
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="locale.title-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;

exports[`Surfaces/Card Primary smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
     data-cy="card"
     style="margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root">
    <div class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;
